pvlib/shading.py

```diff
@@ -0,0 +1,140 @@
+"""
+The ``shading`` module contains functions that model module shading and the
+associated effects on PV module output
+"""
+
+import numpy as np
+import pandas as pd
+
+
+def passias_masking_angle(surface_tilt, gcr):
+    r"""
+    The average masking angle over the slant height of a row.
+
+    The masking angle is the angle from horizontal where the sky dome is
+    blocked by the row in front. The masking angle is larger near the lower
+    edge of a row than near the upper edge. This function calculates the
+    average masking angle as described in [1]_.
+
+    Parameters
+    ----------
+    surface_tilt : numeric
+        Panel tilt from horizontal [degrees].
+
+    gcr : float
+        The ground coverage ratio of the array [unitless].
+
+    Returns
+    ----------
+    mask_angle : numeric
+        Average angle from horizontal where diffuse light is blocked by the
+        preceding row [degrees].
+
+    See Also
+    --------
+    passias_sky_diffuse
+
+    Notes
+    -----
+    The pvlib-python authors believe that Eqn. 9 in [1]_ is incorrect.
```

Comment:
The equation below isn't equivalent to the published equation? Or, you couldn't reproduce the published equation [9]?

Reply:
The latter.  The published equation [9] also gives different results from evaluating the integral numerically: https://gist.github.com/kanderso-nrel/2c6c3a1853338cdef5b4bbc67092ccc8

Reply:
I'm persuaded. Your solution looks correct to me. I computed a few points on the k=1 and k=2 curves to check.